src/Microsoft.AspNet.Diagnostics/ErrorHandlerMiddleware.cs

```diff
@@ -60,7 +61,16 @@ public ErrorHandlerMiddleware(RequestDelegate next, ILoggerFactory loggerFactory
                     context.Response.Headers.Clear();
                     context.Response.OnStarting(_clearCacheHeadersDelegate, context.Response);
 
-                    // TODO: Try clearing any buffered data. The buffering feature/middleware has not been designed yet.
+                    // if buffering is enabled, then clear it as data could have been written into it.
+                    var bufferingFeature = context.GetFeature<IHttpBufferingFeature>();
+                    if (bufferingFeature != null)
+                    {
+                        if (context.Response.Body.CanSeek)
```

Comment:
I wonder if we should even check for the presence of `IHttpBufferingFeature` and just do this check?